### ISSUE TITLE
[apex] ApexDoc rule doesn't match published Salesforce ApexDoc specification

### ISSUE THREAD

**Issue by mitchspano** [pmd/pmd#6189]
**Affects PMD Version:** 7.17.0

**Rule:** ApexDocRule

Please provide the rule name and a link to the rule documentation:
[Apex Documentation](https://docs.pmd-code.org/latest/pmd_rules_apex_documentation.html)

**Description:**
The ApexDocRule in PMD for Apex was validating an incorrect ApexDoc format. It checked for a `@description` tag, but the newly published Salesforce ApexDoc spec doesn't use this tag.

**Code Sample demonstrating the issue:**
The rule expected
```java
/**
 * @description Creates a new Account sObject...
 * @param accountName The account name
 * @return The created Account
 */
public Account createAccount(String accountName) { ... }
```

According to the official [Salesforce ApexDoc format specification](https://developer.salesforce.com/docs/atlas.en-us.258.0.apexcode.meta/apexcode/apex_doc_format.htm), the main description should be text that appears before any ApexDoc tags (like `@param`, `@return`, `@throws`, etc.), without requiring a separate `@description` tag:

**Expected outcome:**
This should be fine:
```java
/**
 * Creates a new Account sObject with the given name and industry.
 * @param accountName The desired name for the new account. Cannot be null or empty.
 * @param industry The industry classification for the new account.
 * @return The newly created Account sObject with its ID populated.
 * @throws AccountManager.AccountException if accountName is invalid
 * or if DML operation fails.
 */
public Account createAccount(String accountName, String industry) { ... }
```
PMD reports a violation - `missing @description`, but it should not.

**Running PMD through:** All mechanisms.
